R/server.R

```diff
@@ -1,110 +1,165 @@
-server <- function(input, output, session) {
-  
-  icd10_usage <- codeusage::icd10_usage
-  snomed_usage <- codeusage::snomed_usage
-  
-  # Reactive expression for the selected dataset
+#' The application server-side
+#'
+#' @param input,output,session Internal parameters for {shiny}.
+#' @noRd
+#' @import shiny
+#' @import bslib
+#' @import bsicons
+#' @import dplyr
+#' @import ggplot2
+#' @importFrom scales comma label_date_short label_comma
+#' @importFrom lubridate month year
+#' @importFrom DT renderDT datatable
+#' @importFrom plotly renderPlotly ggplotly plot_ly config add_lines layout
+#' @import here
+
+app_server <- function(input, output, session) {
+  # DATA: Selected code usage dataset
   selected_data <- reactive({
     if (input$dataset == "snomed") {
-      snomed_usage %>%
-        select(start_date, code = snomed_concept_id, description, usage)
+      codeusage::snomed_usage %>%
+        select(start_date, end_date, code = snomed_concept_id, description, usage)
     } else if (input$dataset == "icd10") {
-      icd10_usage %>%
-        select(start_date, code = icd10_code, description, usage)
+      codeusage::icd10_usage %>%
+        select(start_date, end_date, code = icd10_code, description, usage)
+    } else if (input$dataset == "opcs") {
+      codeusage::opcs_usage %>%
+        select(start_date, end_date, code = opcs_code, description, usage)
     }
   })
-  
-  # Update selectizeInput choices
+
+  # Update choices
   observe({
     updateSelectizeInput(
       session, "code_search",
       choices = unique(selected_data()$code),
-      server = TRUE)
+      server = TRUE
+    )
   })
-  
-  # Filtered data
+
+  # DATA: OpenCodelist
+  selected_codelist <- reactive({
+    req(input$codelist_slug)
+
+    tryCatch(
+      {
+        codelist_s7 <- get_codelist(input$codelist_slug)
```

Comment:
Great to have this processing outside of the data filter. How does R know `get_codelist` function? I can't see it defined elsewehre in the package.

Reply:
It's part of the package, hidden at the bottom of `R/opencodelists.R`  https://github.com/ebmdatalab/codeusage/blob/bc4ffc18fefbcb27c2d1d22565f21d4e13f5b5c1/R/opencodelists.R#L28-L74